public/app/features/alerting/unified/components/contact-points/ContactPointHeader.tsx

```diff
@@ -26,21 +34,54 @@ interface ContactPointHeaderProps {
 export const ContactPointHeader = ({ contactPoint, disabled = false, onDelete }: ContactPointHeaderProps) => {
   const { name, id, provisioned, policies = [] } = contactPoint;
   const styles = useStyles2(getStyles);
+  const [showPermissionsDrawer, setShowPermissionsDrawer] = useState(false);
+  const { selectedAlertmanager } = useAlertmanager();
 
   const [exportSupported, exportAllowed] = useAlertmanagerAbility(AlertmanagerAction.ExportContactPoint);
   const [editSupported, editAllowed] = useAlertmanagerAbility(AlertmanagerAction.UpdateContactPoint);
   const [deleteSupported, deleteAllowed] = useAlertmanagerAbility(AlertmanagerAction.UpdateContactPoint);
-
   const [ExportDrawer, openExportDrawer] = useExportContactPoint();
 
-  const numberOfPolicies = policies.length;
-  const isReferencedByAnyPolicy = numberOfPolicies > 0;
-  const isReferencedByRegularPolicies = policies.some((ref) => ref.route.type !== 'auto-generated');
+  const showManagePermissions = showManageContactPointPermissions(selectedAlertmanager!, contactPoint);
+
+  const regularPolicyReferences = policies.filter((ref) => ref.route.type !== 'auto-generated');
+
+  /**
+   * Number of (regular) policies that reference this contact point -
+   * either taken from the annotations (k8s) or from the alertmanager config
+   */
+  const numberOfPolicies = Number(
+    contactPoint.metadata?.annotations?.[ANNOTATION_INUSE_ROUTES] ?? regularPolicyReferences.length
+  );
+
+  /** Number of rules that use this contact point for simplified routing */
+  const numberOfRules = Number(contactPoint.metadata?.annotations?.[ANNOTATION_INUSE_RULES]) || 0;
```

Comment:
are we going to show zero rules in case we don't get this `contactPoint.metadata?.annotations?.[ANNOTATION_INUSE_RULES]`?

Reply:
If this ends up being 0 then the text won't show in the header at all, as we have a falsy check there :) unless I'm misunderstanding your question?

Reply:
sorry, you're right!, I missed this [line](https://github.com/grafana/grafana/pull/92885/files#diff-bd80acfd65ce0ebbe81500cff314e3c907861d173d3268a0ff86e532ea48f55fR191)